aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2180,7 +2181,7 @@ impl AptosVM {
     ///
     /// This can be useful for off-chain applications that wants to perform additional
     /// measurements or analysis while preserving the production gas behavior.
-    pub fn execute_user_transaction_with_modified_gas_meter<'a, G, F>(
+    pub fn execute_user_transaction_with_modified_gas_meter<'a, G, M, F>(
```

Comment:
Should not we update callers of this function to use new extended algebra?

Reply:
Seems like we don't need to do this explicitly -- the callers already receive a gas meter using `ExtendedMemoryAlgebra`. I don't fully understand why this is the case though. I suspect it may have something do so with some obscure type inference rules.